### PR TITLE
MRG: Use numpydoc from pip

### LINKED ISSUE #3262: Missing sphinx extensions
To get my make working I had to install sphinx-gallery and sphinx_bootstrap_theme. I don't see a reference to them in the contributing documentation, but it seems they should be under the list of development related tools.
Should I add them to the list of python libraries?

### PR DESCRIPTION
**Pull request by Eric89GXL**
Gets rid of our `numpydoc` in favor of the one on `pip`. Also fixes a few problems with our docstrings.

Closes #3262.

### CODE REVIEW

File: doc/contributing.rst [hunk 1]
```diff
@@ -113,30 +125,60 @@ General code guidelines
      >>> run_tests_if_main()
 
   For more details see troubleshooting_.
-
+  
 * Update relevant documentation. Update :doc:`whats_new.rst <whats_new>` for new features and :doc:`python_reference.rst <python_reference>` for new classes and standalone functions. :doc:`whats_new.rst <whats_new>` is organized in chronological order with the last feature at the end of the document.
 
- To ensure that these files were rendered correctly, run the following command::
+Checking and building documenation
```
Comment (agramfort) [inline doc/contributing.rst]: Documentation

File: doc/contributing.rst [hunk 2]
```diff
@@ -113,30 +125,60 @@ General code guidelines
...
+  * numpydoc
+
+Then to build the documentation locally, within the ``mne/doc`` directory do:
+
+.. code-block:: bash
+
+    $ make html-noplot
+
+This will build the docs without building all the examples, which can save
+some time.If you are working on examples or tutorials, you can build specific
```
Comment (agramfort) [inline doc/contributing.rst]: Space before if

**Comment by agramfort** [comment]
Besides lgtm

**Comment by Eric89GXL** [comment]
Comments addressed

**Comment by agramfort** [comment]
Thx

**Pull request merged**